File: VB/MvcPivotChartIntegration/Helper/PivotGridSettingsHelper.vb
Imports DevExpress.Web.Mvc
Imports DevExpress.XtraPivotGrid
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace MvcApplication1.Helper
	Public Class PivotGridSettingsHelper
		Public Shared Sub CustomizePivotGridSettings(ByVal settings As PivotGridSettings)
			settings.Name = "PivotGrid"
			settings.CallbackRouteValues = New With {Key .Controller = "Home", Key .Action = "PivotGridPartial"}

			settings.OptionsChartDataSource.ProvideDataByColumns = False
			settings.ClientSideEvents.EndCallback = "PivotEndCallback"

			settings.BeforeGetCallbackResult = Sub(sender, e)
				Dim pivot As MVCxPivotGrid = CType(sender, MVCxPivotGrid)
				pivot.JSProperties("cpLayoutUpdated") = False
			End Sub
			settings.GridLayout = Sub(sender, e)
				Dim pivot As MVCxPivotGrid = CType(sender, MVCxPivotGrid)
				pivot.JSProperties("cpLayoutUpdated") = True
			End Sub

			settings.Fields.Add(Sub(field)
				field.ID = "fieldDate"
				field.Area = PivotArea.ColumnArea
				field.FieldName = "Date"
				field.GroupInterval = PivotGroupInterval.Date
				field.Caption = "Date"
			End Sub)
			settings.Fields.Add(Sub(field)
				field.ID = "fieldName"
				field.Area = PivotArea.RowArea
				field.FieldName = "Name"
				field.Caption = "Name"
			End Sub)
			settings.Fields.Add(Sub(field)
				field.ID = "fieldValue"
				field.Area = PivotArea.DataArea
				field.FieldName = "Value"
				field.Caption = "Value"
			End Sub)

		End Sub
		Public Shared Function CreatePivotGridSettings() As PivotGridSettings
			Dim settings As New PivotGridSettings()
			CustomizePivotGridSettings(settings)
			Return settings
		End Function
	End Class
End Namespace

File: CS/MvcPivotChartIntegration/Helper/PivotGridSettingsHelper.cs
using DevExpress.Web.Mvc;
using DevExpress.XtraPivotGrid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace MvcApplication1.Helper {
    public class PivotGridSettingsHelper {
        public static void CustomizePivotGridSettings(PivotGridSettings settings)
        {
            settings.Name = "PivotGrid";
            settings.CallbackRouteValues = new { Controller = "Home", Action = "PivotGridPartial" };

            settings.OptionsChartDataSource.ProvideDataByColumns = false;
            settings.ClientSideEvents.EndCallback = "PivotEndCallback";
            
            settings.BeforeGetCallbackResult = (sender, e) => {
                MVCxPivotGrid pivot = (MVCxPivotGrid)sender;
                pivot.JSProperties["cpLayoutUpdated"] = false;
            };
            settings.GridLayout = (sender, e) => {
                MVCxPivotGrid pivot = (MVCxPivotGrid)sender;
                pivot.JSProperties["cpLayoutUpdated"] = true;
            };

            settings.Fields.Add(field =>
            {
                field.ID = "fieldDate";
                field.Area = PivotArea.ColumnArea;
                field.FieldName = "Date";
                field.GroupInterval = PivotGroupInterval.Date;
                field.Caption = "Date";
            });
            settings.Fields.Add(field =>
            {
                field.ID = "fieldName";
                field.Area = PivotArea.RowArea;
                field.FieldName = "Name";
                field.Caption = "Name";
            });
            settings.Fields.Add(field =>
            {
                field.ID = "fieldValue";
                field.Area = PivotArea.DataArea;
                field.FieldName = "Value";
                field.Caption = "Value";
            });
            
        }
        public static PivotGridSettings CreatePivotGridSettings() {
            PivotGridSettings settings = new PivotGridSettings();
            CustomizePivotGridSettings(settings);
            return settings;
        }
    }
}